### PR TITLE
[Update] add developer guide to master branch

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -18,6 +18,65 @@
   - [快速上手 HTAP](/quick-start-with-htap.md)
   - [SQL 基本操作](/basic-sql-operations.md)
   - [深入探索 HTAP](/explore-htap.md)
+- 应用开发
+  - 快速开始
+    - [使用 TiDB Cloud (DevTier) 构建 TiDB 集群](/develop/build-cluster-in-cloud.md)
+    - [使用 TiDB 的增删改查 SQL](/develop/tidb-crud-sql.md)
+    - TiDB 的简单 CRUD 应用程序
+      - [Java](/develop/sample-application-java.md)
+  - 开发者手册
```
Comment (qiancai) [inline TOC.md:27]: 目前 “开发者手册”放在这里要有些困惑它和其他章节的关系，是否可以考虑将 “概览”挪到快速开始前面，将“示例程序”专门设置一个大的章节呢？

**Review by Oreoxmt — APPROVED** [review]

File: TOC.md (hunk 1 above)
Comment (Icemap) [reply TOC.md]: 改了。

**Review by qiancai — APPROVED** [review]

**Pull request merged**